model/patch/cli_intent.go

```diff
@@ -205,22 +222,18 @@ func NewCliIntent(user, project, baseHash, module, patchContent, description str
 	return &cliIntent{
 		DocumentID:    mgobson.NewObjectId().Hex(),
 		IntentType:    CliIntentType,
-		PatchContent:  patchContent,
-		Description:   description,
-		BuildVariants: variants,
-		Tasks:         tasks,
-		SyncAtEndOpts: SyncAtEndOptions{
-			BuildVariants: syncBVs,
-			Tasks:         syncTasks,
-			Statuses:      syncStatuses,
-			Timeout:       syncTimeout,
-		},
-		User:      user,
-		ProjectID: project,
-		BaseHash:  baseHash,
-		Finalize:  finalize,
-		Module:    module,
-		Alias:     alias,
+		PatchContent:  params.PatchContent,
+		Description:   params.Description,
+		BuildVariants: params.Variants,
+		Tasks:         params.Tasks,
+		SyncAtEndOpts: params.SyncParams,
+		User:          params.User,
+		ProjectID:     params.Project,
+		BaseHash:      params.BaseGitHash,
+		Finalize:      params.Finalize,
+		Module:        params.Module,
+		Alias:         params.Alias,
+		IsMbox:        len(params.PatchContent) == 0 || IsMailboxDiff(params.PatchContent),
```

Comment:
I'm having trouble searching, where is IsMbox used for CLIIntent? Do we need this field to be saved since it's just an operation on PatchContent which is already saved?

Reply:
My thinking here is to avoid having to pull the patch content back out of the db when we're processing the patch.
(The intent's `Insert()` function writes the patchContents to gridFS and swaps the `PatchContents` for a `PatchFileID`)

The IsMbox field in the intent is used when we create a new patch from the intent.

Reply:
Oh I see, thanks!